tasks/indexing/task_ipni.go

```diff
@@ -94,6 +97,10 @@ func (I *IPNITask) Do(taskID harmonytask.TaskID, stillOwned func() bool) (done b
 
 	task := tasks[0]
 
+	if task.Complete {
+		return true, nil
+	}
```

Comment:
I don't quite understand what the purpose of this is. Just being careful around retry?

Reply:
Yes. Being extra careful.